### DISCUSSION TITLE
Keycloak Admin Client - User group always null

### DISCUSSION THREAD

**Discussion by denisgmarques** [keycloak/keycloak#23674] [Q&A]
Hello,

  I am using keycloak-admin-client 19.0.3 in a spring boot application an it works in general but I have been struggling to get user groups. If I get group members it works. But when I get some user, the groups property is always null for any user.

   This image is a debug console, we can see that group members was retrieved but on any user, groups property is always null?
![image](https://github.com/keycloak/keycloak/assets/71543/06373928-15dd-4c67-9fce-db18fbcde52e)

   I tried many keycloak-admin-client versions.

   I appreciate any help.

Tks.

**Comment by danielFesenmeyer** [comment]
The field UserRepresentation#groups is only used for import and exports of users, in any other context it is not relevant currently. That's why it's null in this case.

**Reply by denisgmarques** [reply]
Hello @danielFesenmeyer,

I think the relevance depends on the client requirements, in my case to know the user groups is relevant and I need to find some solution for this case. Do you know some workaround for this issue? I didn't tried the keycloak-admin-client for nodeJs, maybe it works different. This limitation is on keycloak-admin-client for spring-boot or is a limitation in Keycloak admin rest api?

**Reply by danielFesenmeyer** [reply]
I just had a look at the current backend code (admin rest api) on `main`: https://github.com/keycloak/keycloak/blob/58131f1dccb7113cb708407d6cf90842216c8d00/services/src/main/java/org/keycloak/services/resources/admin/UserResource.java#L318
There, the groups are just not mapped. (And I know from experience, that groups are not mapped in that context, at least for the latest keycloak versions.) I guess this limitation is due to performance reasons.

From my POV, there are currently these workarounds:
- get the group memberships in a separate call
- create a custom endpoint which also maps the groups

**Reply by denisgmarques** [reply]
Hello,

Using the Keycloak Rest Api directly I could get full data for groups and users passing this query param in the requests: 

```
?briefRepresentation=false
```

For example, the following request I did against group api to get all groups including the group attributes, that is a king of property that keycloak-admin-client returns null:
```
GET KCHOST:KCPORT/admin/realms/<MYREALM>/groups?briefRepresentation=false
```

The question now is: 

Is it possible to configure keycloak-admin-client to pass this query param briefRepresentation=false automatically?

I was looking the docs and there is some methods where I can pass briefRepresentation, I will try!

https://www.keycloak.org/docs-api/21.0.1/javadocs/org/keycloak/admin/client/resource/UsersResource.html

Tks

**Comment by kwame-mintah** [comment]
Hi @denisgmarques, I was able to get the users' group like this in the past. Could you try this snippet below?

```java
keycloak.realm(realm).users().get("user-id").groups().stream().map(GroupRepresentation::getName).findFirst().map(Object::toString).orElse("NO_GROUP")
```
Would link the [javadocs](https://javadoc.io/doc/org.keycloak), but the site is currently down for me. Apologies in advance if I have misunderstood the question.

**Reply by denisgmarques** [reply]
Hello,

Your suggestion solves part of the problem, it works.

Thank you
